tap_mysql/__init__.py

```diff
@@ -82,13 +82,48 @@ class InputException(Exception):
 
 @attr.s
 class StreamState(object):
-    stream = attr.ib()
-    replication_key = attr.ib()
-    replication_key_value = attr.ib()
+    '''Represents the state for a single stream.
+
+    The state for a stream consists of four properties:
+
+      * tap_stream_id - the identifier for the stream
+      * replication_key (optional, string) - name of the field used for
+        bookmarking
+      * replication_key_value (optional, string or int) - current value of
+        the bookmark
+      * version (optional, int) - the version number of the stream.
```

Comment:
@mdelaurentis this is marked as `optional` -- in what cases would a stream state exist without a version?